spec/features/pictures_feature_spec.rb

```diff
@@ -0,0 +1,92 @@
+require 'rails_helper'
+
+feature 'pictures' do
+  context 'no pictures have been added' do
+    scenario 'should display a prompt to add a picture' do
+      visit '/pictures'
+      expect(page).to have_content 'No pictures yet!'
+      expect(page).to have_link 'Click here to add a picture!'
+    end
+  end
+
+  context 'uploading a picture' do
+    scenario 'succeeds when a title is entered' do
+      sign_up
+      upload_picture
+      expect(page).to have_css("img[src*='Pompey.jpg']")
+    end
+
+    scenario 'fails without a title' do
+    sign_up
+    visit '/pictures'
+    click_link 'Click here to add a picture!'
+    attach_file "Image", 'spec/features/Pompey.jpg'
+    click_button 'Upload picture'
+    expect(page).to have_content 'error'
+    end
+
+    scenario 'does not succeed without an image file' do
+      sign_up
+      visit '/pictures'
+      click_link 'Click here to add a picture!'
+      fill_in 'Title', with: 'nothing'
+      click_button 'Upload picture'
+      expect(page).to have_content 'error'
+    end
+
+  end
+
+  context 'viewing a picture' do
+    let!(:pfc) {Picture.create(title:'PFC',
+                               image: File.new("spec/features/Pompey.jpg") )}
```

Comment:
Expression at 41, 77 should be on its own line.<br>Space inside parentheses detected.